addons/mail/static/src/models/follower/follower.js

```diff
@@ -89,6 +89,7 @@ function factory(dependencies) {
                 method: 'message_unsubscribe',
                 args: [[this.followedThread.id], partner_ids, channel_ids]
             }));
+            this.followedThread.fetchAndUpdateSuggestedRecipients();
```

Comment:
It's safer to move this call after the delete, to keep the delete as close as possible (in code but also in terms of async execution) to the return of the previous rpc.